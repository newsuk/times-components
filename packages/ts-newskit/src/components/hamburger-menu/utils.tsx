import React from 'react';
import { Story } from '@storybook/addon-docs';
import { ThemeProvider } from 'newskit';
import { TimesWebLightTheme } from '../../theme';

export const StoryWithTheme: React.FC<{ name: string }> = ({
  name,
  children
}) => (
  <ThemeProvider theme={TimesWebLightTheme}>
    <Story name={name}>{children}</Story>
  </ThemeProvider>
);
