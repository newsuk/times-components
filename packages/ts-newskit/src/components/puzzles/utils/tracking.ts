import { PuzzleScrollClickHandlerType, MouseEventType } from '../cards-container/types';

type TrackingPuzzleCategoryProps = {
  nextPrev: string;
  puzzleType: string;
};

export const puzzleCategoryClickTracking = (
  event: MouseEventType,
  puzzleScroll: TrackingPuzzleCategoryProps,
  clickHandler: PuzzleScrollClickHandlerType
) => {
  puzzleScroll && clickHandler(event, puzzleScroll);
};
