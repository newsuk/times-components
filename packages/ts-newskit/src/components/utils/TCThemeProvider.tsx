import React from 'react';
import { NewsKitProvider, Theme } from 'newskit';
import { TimesWebLightTheme } from '../../theme';

type TCThemeProviderProps = {
  theme?: Theme;
};

export const TCThemeProvider: React.FC<TCThemeProviderProps> = ({
  theme = TimesWebLightTheme,
  children
}) => <NewsKitProvider theme={theme}>{children}</NewsKitProvider>;
