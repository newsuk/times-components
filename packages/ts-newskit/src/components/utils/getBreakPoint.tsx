import { useBreakpointKey } from 'newskit';
import { SecondaryMenuItem } from '../secondary-menu/types';

export const getBreakpoint = (data: SecondaryMenuItem[]) => {
  const breakpointKey = useBreakpointKey();

  const menuItems =
    breakpointKey === 'md' ? 7 : breakpointKey === 'lg' ? 9 : data.length;

  const moreMenuLength = data.length - menuItems;

  return { moreMenuLength, menuItems, breakpointKey };
};
