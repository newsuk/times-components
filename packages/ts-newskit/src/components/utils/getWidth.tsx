export const getWidth = (el: any) => el.clientWidth;
