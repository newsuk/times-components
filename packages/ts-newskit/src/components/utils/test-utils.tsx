import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from 'newskit';
import { TimesWebLightTheme } from '../../theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={TimesWebLightTheme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
