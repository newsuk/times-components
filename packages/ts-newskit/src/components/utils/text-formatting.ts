import xss from 'xss';

const decodeEntities = (str: string) =>
  str
    .replace(/&#(\d+);/g, (__: string, d: number) => String.fromCharCode(d))
    .replace(/\s+/g, ' ');

export const sanitiseCopy = (copy: string = '', allowedTags: {} = {}) =>
  xss(decodeEntities(copy), {
    whiteList: allowedTags,
    stripIgnoreTag: true,
    stripIgnoreTagBody: ['script']
  });
