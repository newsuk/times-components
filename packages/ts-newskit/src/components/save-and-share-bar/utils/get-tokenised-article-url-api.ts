// @ts-ignore
import { getTokenisedArticleUrl } from '@times-components/provider-queries';
// @ts-ignore
import { makeClient } from '@times-components/utils';

const client = makeClient();

export const getTokenisedArticleUrlApi = (id: string) =>
  client.query({ query: getTokenisedArticleUrl, variables: { id } });
