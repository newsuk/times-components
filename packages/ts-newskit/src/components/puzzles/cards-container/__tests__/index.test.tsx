import React from 'react';
import { render } from '../../../../utils/test-utils';
import { CardsContainer } from '../index';
import '@testing-library/jest-dom';
import { fireEvent } from '@testing-library/react';
import { puzzleGame } from './../fixtures/puzzleGame.json';

describe('CardsContainer tests', () => {
  it('should render a snapshot without scroller', () => {
    const { asFragment } = render(
      <CardsContainer
        cards={Array(8).fill(puzzleGame)}
        title="Crosswords"
        seeAllLink="https://newskit.co.uk/components"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render a snapshot with scroller', () => {
    const { asFragment } = render(
      <CardsContainer
        cards={Array(8).fill(puzzleGame)}
        title="Crosswords"
        seeAllLink="https://newskit.co.uk/components"
        isScrollable={true}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should scroll right once scroll right control is clicked', () => {
    const { getByTestId } = render(
      <CardsContainer
        cards={Array(7).fill(puzzleGame)}
        title="Crosswords"
        seeAllLink="https://newskit.co.uk/components"
        isScrollable={true}
      />
    );
    const scrollContainer = getByTestId('scroll-container');
    const btnScrollRight = getByTestId('scroll-right');

    fireEvent.click(btnScrollRight);
    expect(scrollContainer.scrollLeft).toBeGreaterThan(0);
  });
  it('should scroll left once scroll left control is clicked', () => {
    const { getByTestId } = render(
      <CardsContainer
        cards={Array(7).fill(puzzleGame)}
        title="Crosswords"
        seeAllLink="https://newskit.co.uk/components"
        isScrollable={true}
      />
    );
    const scrollContainer = getByTestId('scroll-container');
    const btnScrollLeft = getByTestId('scroll-left');

    fireEvent.click(btnScrollLeft);
    expect(scrollContainer.scrollLeft).toBeLessThan(0);
  });
});
