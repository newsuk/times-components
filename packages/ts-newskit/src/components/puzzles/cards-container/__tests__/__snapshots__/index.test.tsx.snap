// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsContainer tests should render a snapshot with scroller 1`] = `
<DocumentFragment>
  <div
    class="css-1y25n6u"
    data-testid="title-bar"
  >
    <div
      class="css-16bai6p"
    >
      <section
        class="css-1oteowz"
      >
        <h2
          class="nk-headline-heading css-sxktxw"
        >
          Crosswords
        </h2>
      </section>
    </div>
    <div
      class="css-11l110t"
    >
      <div
        class="css-17x5lw"
      >
        <a
          class="css-1tarvo2"
          data-testid="buttonLink"
          href="https://newskit.co.uk/components"
        >
          <span
            class="css-n79ak9"
          >
            SEE ALL
          </span>
        </a>
        <button
          class="css-1llax8i"
          data-testid="scroll-left"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="css-1arce68-EmotionIconBase ex0cdmw0"
            fill="currentColor"
            focusable="false"
            overrides="[object Object]"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <polygon
              points="17.77,3.77 16,2 6,12 16,22 17.77,20.23 9.54,12"
            />
          </svg>
        </button>
        <button
          class="css-13tcow"
          data-testid="scroll-right"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="css-1arce68-EmotionIconBase ex0cdmw0"
            fill="currentColor"
            focusable="false"
            overrides="[object Object]"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    class="css-1r18k3i"
  >
    <div
      class="css-6uqogc"
      data-testid="scroll-container"
      tabindex="0"
    >
      <div
        class="css-ocffxe"
      >
         
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardsContainer tests should render a snapshot without scroller 1`] = `
<DocumentFragment>
  <div
    class="css-1y25n6u"
    data-testid="title-bar"
  >
    <div
      class="css-16bai6p"
    >
      <section
        class="css-1oteowz"
      >
        <h2
          class="nk-headline-heading css-sxktxw"
        >
          Crosswords
        </h2>
      </section>
    </div>
    <div
      class="css-11l110t"
    />
  </div>
  <div
    class="css-1r18k3i"
  >
    <div
      class="css-6uqogc"
      data-testid="scroll-container"
      tabindex="0"
    >
      <div
        class="css-13ctbov"
      >
         
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
        <div
          class="css-1kxmbtt"
          data-test-id="single-card"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
